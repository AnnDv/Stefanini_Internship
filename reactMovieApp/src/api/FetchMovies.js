import { CONFIG } from "../config";
import React, { useEffect, useState } from "react";
import MovieCard from "../components/MovieCard";

const API_URL = CONFIG.BASE_URL + CONFIG.POPULAR_MOVIE_URL + CONFIG.API_KEY;

export const FetchMovies = () => {
  const [movies, setMovies] = useState([]);

  useEffect(() => {
    fetch(API_URL)
    .then(response => response.json())
    .then(data => { 
      console.log(data.results);
      setMovies(data.results);
    });
  }, [])
  
  return (
    <div className='container'>
      <MovieCard movies={movies}/>    
    </div>
      
  );
}

export default FetchMovies;