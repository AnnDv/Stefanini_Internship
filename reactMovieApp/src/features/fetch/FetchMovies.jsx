import React, { useEffect, useState } from 'react';
import MovieList from '../movieList/MovieList.jsx';
import SearchBox from '../search/SearchMovie.jsx';
import { useGetPopularMoviesQuery, useLazyGetSearchMovieQuery } from '../../store/sliceApi';
// import Debouncer from '../../components/Debounce/Debouncer';

const FetchMovies = () => {
  const { data: movie } = useGetPopularMoviesQuery();
  const [movies, setMovies] = useState([]);
  const [search, setSearch] = useState('');
  const [execute, { data: list }] = useLazyGetSearchMovieQuery();

  // const debouncedSearchTerm = Debouncer(search, 500);

  useEffect(() => {
    if (!search.length) {
      setMovies(movie);
    } else if (search.length >= 3) {
      execute(search);
      setMovies(list);
    }
  }, [execute, search, list, setMovies, movie]);

  return (
    <>
      <nav className='navbar'>
        <h3>Movie TMDB</h3>
          <SearchBox
          search={search}
          setSearch={setSearch}
        />
      </nav>
      <div data-testid="fetch" className='container'>
        <MovieList movies={movies}/>
      </div>
    </>
  );
};

export default FetchMovies;
