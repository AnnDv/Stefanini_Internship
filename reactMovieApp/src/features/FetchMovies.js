import { getData } from "../api/getData";
import { CONFIG } from "../config";
import MovieCard from "../components/MovieCard";
import  React, { useEffect, useState } from "react";

const API_URL = CONFIG.BASE_URL + CONFIG.POPULAR_MOVIE_URL + CONFIG.API_KEY;

const FetchMovies = () => {
  const [movies, setMovies] = useState([]);

  useEffect(() => {
    getData(API_URL).then((result) => { 
    setMovies(result)
  })
}, [])

  console.log(movies)

  return (
    <div className='container'>
      <MovieCard movies={movies}/>    
    </div>
  );
}

export default FetchMovies;