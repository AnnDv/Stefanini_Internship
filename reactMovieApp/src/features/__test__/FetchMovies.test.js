import React from 'react';
import { render, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import FetchMovies from '../FetchMovies';
import mockData from '../../api/getData';

jest.mock('../../api/getData');

describe('FetchMovies', () => {
  it('should add movie data', async () => {
    mockData.mockResolvedValue([{
      title: 'Black', overview: 'text',
    }]);
    const { getByTestId, baseElement } = render(<FetchMovies />);
    await waitFor(async () => {
      const movieColl = getByTestId('fetch').children[0].children[0];
      const movieCollLength = movieColl.childElementCount;
      expect(movieCollLength).toBeGreaterThanOrEqual(1);
    });
    expect(baseElement).toMatchSnapshot();
  });

  it('should fail if we have an empty array', async () => {
    console.log(mockData);
    mockData.mockResolvedValue([]);
    const { getByTestId, baseElement } = render(<FetchMovies />);
    await waitFor(async () => {
      const movieColl = getByTestId('fetch').children[1].children[0];
      const movieCollLength = movieColl.childElementCount;
      expect(movieCollLength).toBeGreaterThan(0);
    });
    expect(baseElement).toMatchSnapshot();
  });
});
