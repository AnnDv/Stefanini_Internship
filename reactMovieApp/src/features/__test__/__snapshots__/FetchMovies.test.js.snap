// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FetchMovies should add movie data 1`] = `
<body>
  <div>
    <nav
      class="navbar"
    >
      <h3>
        Popular Movies:
      </h3>
    </nav>
    <div
      class="container"
      data-testid="fetch"
    >
      <div
        class="grid"
        data-testid="list"
      >
        <div
          class="card_description"
          data-testid="m-undefined"
        >
          <div>
            <div>
              <img
                alt="movie"
                class="movie_img"
                src="https://image.tmdb.org/t/p/w500undefined"
              />
            </div>
            <div
              class="title"
            >
              <p>
                Black
              </p>
            </div>
            <div
              class="card-body"
            >
              <button
                class="button"
              >
                View More
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
